src/cli/main.cpp

```diff
@@ -415,49 +418,100 @@ void tryListReadWriteRegsOfSingleNodeScenario(Executor&                   execut
+            spdlog::warn("Failed to receive message (err={}).", *failure);
             return;
         }
-        const auto raw_msg = cetl::get<Receive::Success>(std::move(raw_msg_result));
+        const auto receive = cetl::get<Receive::Success<Heartbeat>>(rcv_result);
+        logHeartbeat(receive.message, receive.publisher_node_id, receive.priority);
+    }
+}
+
+/// Demo of daemon's publisher - publishes `uavcan::time::Synchronization_1_0` messages (every ~1s during 30s).
+/// The first half with low priority, and the second half with high priority.
+///
+void tryPublisherScenario(Executor& executor, cetl::pmr::memory_resource& memory, const Daemon::Ptr& daemon)
+{
+    using ocvsmd::sdk::Publisher;
+    using SyncMessage   = uavcan::time::Synchronization_1_0;
+    using MakePublisher = Daemon::MakePublisher;
+
+    spdlog::info("tryPublisherScenario -----------------");
 
-        Heartbeat heartbeat{&memory};
-        if (!tryDeserializePayload({raw_msg.data.get(), raw_msg.size}, heartbeat))
+    auto pub_sender = daemon->makePublisher(SyncMessage::_traits_::FixedPortId);
+    auto pub_result = sync_wait<MakePublisher::Result>(executor, std::move(pub_sender), 2s);
+    if (const auto* const failure = cetl::get_if<MakePublisher::Failure>(&pub_result))
+    {
+        spdlog::error("Failed to make publisher (err={}).", *failure);
+        return;
+    }
+    const auto publisher = cetl::get<MakePublisher::Success>(std::move(pub_result));
+
+    if (const auto opt_error = publisher->setPriority(ocvsmd::sdk::CyphalPriority::Low))
+    {
+        spdlog::warn("Failed to set 'low' priority publisher (err={}).", *opt_error);
+    }
+
+    uavcan::time::Synchronization_1_0 sync_msg{&memory};
+
+    int           counter       = 0;
+    constexpr int duration_secs = 30;
+    spdlog::info("Publishing time syncs for {} secs...", duration_secs);
+    const auto until_timepoint = executor.now() + std::chrono::seconds{duration_secs};
+    while (until_timepoint > executor.now())
+    {
+        ++counter;
+        if (counter == 15)
         {
-            spdlog::error("Failed to deserialize heartbeat.");
+            if (const auto opt_error = publisher->setPriority(ocvsmd::sdk::CyphalPriority::High))
+            {
+                spdlog::warn("Failed to set 'high' priority publisher (err={}).", *opt_error);
+            }
+        }
+
+        const auto timeout = until_timepoint - executor.now();
+
+        auto sender = publisher->publish(sync_msg, 1s);
+        if (const auto error = sync_wait<OptError>(executor, std::move(sender), timeout))
+        {
+            spdlog::warn("Failed to publish message (err={}).", *error);
             return;
         }
-        logHeartbeat(heartbeat, raw_msg.publisher_node_id, raw_msg.priority);
+
+        sync_msg.previous_transmission_timestamp_microsecond =
+            std::chrono::duration_cast<std::chrono::microseconds>(executor.now().time_since_epoch()).count();
```

Comment:
`uavcan::time::Synchronization_1_0` was chosen as just suitable candidate to demo SDK publishing. So, correctness of the `previous_transmission_timestamp_microsecond` field population was not a goal (actually, I doubt that current `libcyphal` api allows proper implementation - currently there is no api to know when exactly a transfer was pushed to the media).

Reply:
At least we could approximate and remember the time sampled during the previous publication:

```c++
std::chrono::microseconds prev_time{};  // this is a field

sync_msg.previous_transmission_timestamp_microsecond = prev_time.count();
prev_time = std::chrono::duration_cast<std::chrono::microseconds>(executor.now().time_since_epoch());
```

Reply:
Yup, this is approximately what I've done already (see line 511). I don't need extra `prev_time` var - my `sync_msg` holds it for the next publishing.